src/tools/rbd/Utils.cc

```diff
@@ -815,6 +815,31 @@ int snap_set(librbd::Image &image, const std::string &snap_name) {
   return 0;
 }
 
+bool calc_sparse_extent(const bufferptr &bp, size_t sparse_size, uint64_t length,
+                       size_t &write_offset, size_t &write_length,
+                       size_t &pos, size_t &offset) {
+  size_t shift;
```

Comment:
do you mean a size actually? I mean, when I see the variable of shift, I guess you are going to do a 'shift', such as ">>" or "<<". but actually you mean a size. I think you can make it more clear.